### PR TITLE
Add .Net Framework API doc

### LINKED ISSUE #2457: API Reference for .Net
We have it for C, Java, JS and Python, we should also cover .Net ...

### PR DESCRIPTION
**Pull request by lissyx**
Fixes #2457

### CODE REVIEW

File: doc/DotNet-API.rst [hunk 1]
```diff
@@ -0,0 +1,29 @@
+.Net Framework
+==============
+
+DeepSpeech
+----------
+
+.. doxygenclass:: DeepSpeechClient::DeepSpeech
+   :project: deepspeech-dotnet
+   :members:
+
```
Comment (carlfm01) [inline doc/DotNet-API.rst]: @lissyx  It would be great if the interface is also included:

`.. doxygenclass:: DeepSpeechClient::IDeepSpeech`

Comment (lissyx) [reply doc/DotNet-API.rst]: That's one of the thing I was unsure about : does it make sense to have both? And why?

**Comment by reuben** [comment]
I defer to Carlos.

File: doc/DotNet-API.rst (hunk 1 above)
Comment (carlfm01) [reply doc/DotNet-API.rst]: @lissyx just FTR for dependency injection, users will now which interface and type are available for registration.

**Pull request merged**